lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -432,10 +446,53 @@ Function Choco-Uninstall
     $result.changed = $true
 }
 
+Function Choco-ConfigureProxy
+{
+    [CmdletBinding()]
+
+    param(
+        [Parameter(Mandatory=$false, Position=1)]
+        [string]$proxy_url,
+        [Parameter(Mandatory=$false, Position=2)]
+        [string]$proxy_username,
+        [Parameter(Mandatory=$false, Position=3)]
+        [string]$proxy_password
+    )
+    $hash = @{
+        proxy = $proxy_url
+        proxyUser = $proxy_username
+        proxyPassword = $proxy_password
+    }
+    foreach ($h in $hash.GetEnumerator()) {
+        if ($($h.Value))
+        {
+          $cmd = "$executable config set $($h.Name) $($h.Value)"
+        }
+        else
+        {
+          $cmd = "$executable config unset $($h.Name)"
+        }
+        $results = Invoke-Expression $cmd
+        if ($LastExitCode -ne 0)
+        {
+            Set-Attr $result "choco_error_cmd" $cmd
```

Comment:
Since `2.3`, we no longer need to use `Set-Attr`. Just do `$result.choco_error_cmd = $cmd` instead.

Reply:
ok (it was written one year ago...)